orchestration/flows/bl832/nersc.py

```diff
@@ -0,0 +1,207 @@
+            logger.error("NERSC credential files are missing.")
+            raise FileNotFoundError("NERSC credential files are missing.")
+
+        try:
+            return NerscClient(client_id_path, sfapi_key_path)
+        except Exception as e:
+            logger.error(f"Failed to create NERSC client: {e}")
+            raise e
+
+    def reconstruct(
+        self,
+        file_path: str = "",
+    ) -> bool:
+        """
+        Use NERSC for tomography reconstruction
+        """
+        logger.info("Starting NERSC reconstruction process.")
+
+        # Can't use this long term in production. Need to find a better way to handle credentials.
+        # Want to run this as the alsdev user
+        username = os.getenv("NERSC_USERNAME")
+        password = os.getenv("NERSC_PASSWORD")
+
+        user = self.client.user()
+
+        home_path = f"/global/homes/{user.name[0]}/{user.name}"
+        scratch_path = f"/pscratch/sd/{user.name[0]}/{user.name}"
+        logger.info(home_path)
+        logger.info(scratch_path)
+
+        image_name = self.config.harbor_images832["recon_image"]
+        path = Path(file_path)
+        folder_name = path.parent.name
+        file_name = path.stem
+
+        # Can't use this long term in production. Need to find a better way to handle credentials.
+        # Want to run this as the alsdev user
+        username = os.getenv("NERSC_USERNAME")
+        password = os.getenv("NERSC_PASSWORD")
+
+        job_script = f"""#!/bin/bash
+            #SBATCH -q preempt
+            #SBATCH -A als
+            #SBATCH -C cpu
+            #SBATCH --job-name=tomorecon_nersc_mpi_hdf5_1-0
+            #SBATCH --output={scratch_path}/nerscClient-test/%x_%j.out
+            #SBATCH --error={scratch_path}/nerscClient-test/%x_%j.err
+            #SBATCH -N 1
+            #SBATCH --ntasks-per-node 1
+            #SBATCH --cpus-per-task 64
+            #SBATCH --time=00:15:00
+            #SBATCH --exclusive
+
+            date
+
+            srun podman-hpc login registry.nersc.gov --username {username} --password {password}
+            srun podman-hpc run
+            --volume {home_path}/tomo_recon_repo/microct/legacy/sfapi_reconstruction.py:/alsuser/sfapi_reconstruction.py
+            --volume {home_path}/tomo_recon_repo/microct/legacy/input.txt:/alsuser/input.txt
+            --volume {scratch_path}/microctdata:/alsdata
+            --volume {scratch_path}/microctdata:/alsuser/ \
+            registry.nersc.gov/als/{image_name} \
+            python sfapi_reconstruction.py {file_name} {folder_name}
+
+            date
+            """
+
+        try:
+            logger.info("Submitting reconstruction job script to Perlmutter.")
+            job = self.client.perlmutter.submit_job(job_script)
+            job.complete()  # waits for job completion
+            logger.info("Reconstruction job completed successfully.")
+            return True
+        except Exception as e:
```

Comment:
This will raise an error if the job is not visible right away as an 'active job' on perlmutter, despite having a job ID:

`11:00:25.867 | ERROR   | orchestration.flows.bl832.job_controller - Failed to submit or complete reconstruction job: Job not found: 33817933`

This is problematic because the job _is_ running (or at least in the queue), and it is important not to run the multires step until recon is complete

Reply:
This issue was fixed by the sfapi_client team: https://github.com/NERSC/sfapi_client/issues/93